v2/core/RecipientHooks.sol

```diff
@@ -4,29 +4,44 @@ pragma solidity >=0.8.19;
 import { ISablierV2LockupRecipient } from "@sablier/v2-core/src/interfaces/hooks/ISablierV2LockupRecipient.sol";
 
 abstract contract RecipientHooks is ISablierV2LockupRecipient {
+    address public immutable sablierLockup;
+
     mapping(address => uint256) internal _balances;
 
+    constructor(address sablierLockup_) {
+        sablierLockup = sablierLockup_;
+    }
+
     // Do something after a stream was canceled by the sender.
     function onStreamCanceled(
         uint256 streamId,
         uint128, /* senderAmount */
         uint128 /* recipientAmount */
     )
         external
-        pure
+        view
     {
+        // Check: the caller is the lockup contract.
+        require(msg.sender == sablierLockup, "unauthorized");
+
         // Liquidate the user's position.
         _liquidate({ nftId: streamId });
     }
 
     // Do something after a stream was renounced by the sender.
-    function onStreamRenounced(uint256 streamId) external pure {
+    function onStreamRenounced(uint256 streamId) external view {
+        // Check: the caller is the lockup contract.
+        require(msg.sender == sablierLockup, "unauthorized");
```

Comment:
let's use custom errors

we want to promote them over `require`

Reply:
Agree. You missed at one place for which I have pushed directly to `main`: https://github.com/sablier-labs/examples/commit/34fea9c3efa516872301a97e93c12dfa98da0d4c.